src/reducer.js

```diff
@@ -52,16 +52,43 @@ function recursiveClean (value, model) {
   let isValueArray = Array.isArray(value)
   let output = isValueArray ? [] : {}
   let iteratorFn = isValueArray ? _.forEach : _.forIn
+
   iteratorFn(value, (subValue, key) => {
-    const notEmpty = !_.isEmpty(subValue) || subValue instanceof File
-    if (Array.isArray(subValue) && notEmpty) {
-      output[key] = recursiveClean(subValue, _.get(model, 'items'))
-    } else if (!(subValue instanceof File) && _.isObject(subValue) && notEmpty) {
-      output[key] = recursiveClean(subValue, _.get(model, 'properties.' + key))
-    } else if (notEmpty || _.isNumber(subValue) || typeof subValue === 'boolean' || subValue instanceof Boolean) {
-      output[key] = subValue
+    const notEmpty = !_.isEmpty(subValue) ||
+      _.isNumber(subValue) ||
+      typeof subValue === 'boolean' ||
+      subValue instanceof Boolean ||
+      subValue instanceof File
+
+    let subModel
+
+    // arrays have their model props nested further than objects
+    if (Array.isArray(subValue)) {
+      subModel = _.get(model, `properties.${key}.items`)
+    // if we are iterating through an array
+    // all of the items will need the current model
+    } else if (isValueArray) {
+      subModel = model
+    // normal object props are nested in properties under the key name
+    } else {
+      subModel = _.get(model, `properties.${key}`)
+    }
+
+    let autoClean = _.get(subModel, 'autoClean')
+
+    // don't clean if the model has autoClean strictly set to false
+    // or if it is not empty
+    if (autoClean === false || notEmpty) {
+      // recur on objects and arrays that are not files
```

Comment:
Thanks for the comment. Usually when I see `typeof something === 'object'`, I'm unsure if the author intended to only compare it only to `objects` or it was deliberate to test against `objects` and `arrays` since it's a common confusion that `arrays` are objects and `typeof` returns it as `object`.

Reply:
Yeah. Javascript is real helpful that way...

Reply:
@sophypal 

> since it's a common confusion that `arrays` are objects and `typeof` returns it as `object`.

`typeof([])` _DOES_ return `object`.  `Array.isArray([])` is how you have to test for arrays.

> The JavaScript Array object is a global object that is used in the construction of arrays; which are high-level, list-like objects.

- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array

Reply:
Yes, agreed. Since @gkchestertron is wanting to test both, and `typeof` satisfies his requirement. He's not wanting to test for arrays separately. It's my opinion to make it explicit that he's testing for both Objects and Arrays since it's more clear that was the intention instead of an oversight into what `typeof([])` returns.

Reply:
@notmessenger 
> since it's a common confusion that arrays are objects and typeof returns it as object.

I wasn't negating the fact that `typeof([])` returns `object`. I was simply expressing that there is confusion around this.

Reply:
That is fair - what is your convention? Array.isArray(thing) || _.isObject(thing) ?

Reply:
I was actually ok with leaving it in since you had a comment indicating your intentions. I had thought that `_.isObject(thing)` would work to test only objects but it returns true for functions and arrays as well. :(

Reply:
oh - I'll leave it then